internal/worker/bootstrap/addressfinder.go

```diff
@@ -49,7 +51,18 @@ func CAASBootstrapAddresses(ctx context.Context, env environs.Environ, bootstrap
 
 // IAASBootstrapAddressFinder returns the bootstrap addresses for IAAS.
 func IAASBootstrapAddressFinder(ctx context.Context, config BootstrapAddressesConfig) (network.ProviderAddresses, error) {
```

Comment:
This could just be changed to use the provider tracker, then you wouldn't need this set of changes.

Reply:
Done & Done